File: cilicili-admin/src/api/room.js
import request from '../utils/request'

export function SaveRoom(room) {
    return request({
        url: '/room-service/create',
        method: 'post',
        data: room
    })
}

export function FindAllRooms() {
    return request({
        url: '/room-service/findAll',
        method: 'get'
    })
}

export function FindAllRoomsByLiving(living) {
    return request({
        url: '/room-service/findByLiving/' + living,
        method: 'get'
    })
}

export function FindAllRoomsByStatus(status) {
    return request({
        url: '/room-service/findByStatus/' + status,
        method: 'get'
    })
}

File: cilicili-go/service/room/room.go
package room

import (
	"cilicili-go/db/room"
	"cilicili-go/model/entity"
)

type BaseRoomServiceInterface interface {
	Create(room *entity.Room)
	FindAll() []*entity.Room
	FindById(id string) *entity.Room
	FindByLiving(living bool) []*entity.Room
	FindByStatus(status int) []*entity.Room
}

var roomDao = new(room.BaseRoomDao)

func (b *BaseRoomService) Create(room *entity.Room) {
	roomDao.Save(room)
}

func (b *BaseRoomService) FindAll() []*entity.Room {
	return roomDao.FindAll()
}

func (b *BaseRoomService) FindById(id string) *entity.Room {
	return roomDao.FindById(id)
}

func (b *BaseRoomService) FindByStatus(status int) []*entity.Room {
	return roomDao.FindByStatus(status)
}

func (b *BaseRoomService) FindByLiving(living bool) []*entity.Room {
	return roomDao.FindByLiving(living)
}

type BaseRoomService struct {
}
